pinot-core/src/main/java/org/apache/pinot/core/query/reduce/BrokerReduceService.java

```diff
@@ -272,7 +274,20 @@ public BrokerResponseNative reduceOnDataTable(BrokerRequest brokerRequest,
     }
 
     QueryContext queryContext = BrokerRequestToQueryContextConverter.convert(brokerRequest);
-    DataTableReducer dataTableReducer = ResultReducerFactory.getResultReducer(queryContext);
+
+    // Assuming all elements in the merged data table are of same type
+    Object internalObject = null;
+
+    if (QueryContextUtils.isDistinctQuery(queryContext)) {
```

Comment:
Hmm, any way to avoid this special casing here? I know we have special-cased DISTINCT in other places, but that is something we want to remove.

Reply:
There should be no change on broker side, server side should return the same response with/without the optimization. If not, it is protocol change and will cause backward incompatible issue